### DISCUSSION TITLE
Error on gradle build

### DISCUSSION THREAD

**Discussion by pablo** [Keyfactor/ejbca-ce#660] [Q&A] [answered]
I am trying to build main branch in my computer:

* macOS Sonoma 14.6.1 (Intel)
* java version "11.0.24" 2024-07-16 LTS
* gradle 8.7

However I am running into this error:

 ```ejbca-ce/modules/edition-specific/src-ejb/org/ejbca/core/ejb/EnterpriseEditionWSBridgeSessionBean.java:31: error: cannot access NoSuchSlotException```

Any ideas?

**Comment by primetomas** [comment, marked as answer]
gradle is initial/beta in the current main. It should be more stable I believe after the next release. I suggest waiting until EJBCA 9.0 to dig deeper into Gradle builds. It shouldn't be too long.

**Reply by pablo** [reply]
Thanks a lot. Built it with ant. No issues.

**Reply by primetomas** [reply]
Yes, ant should be no problems. Gradle will probably come more in the future.

**Comment by SaeidKazemi78** [comment]
Now , i have cloned the 8.3.2 version of ejbca-ce , and when i want to build it , igot error 

` error: cannot access NoSuchSlotException
public class EnterpriseEditionWSBridgeSessionBean implements EnterpriseEditionWSBridgeSessionLocal {
       ^
  class file for com.keyfactor.util.keys.token.pkcs11.NoSuchSlotException not found`


what should i do ? is which version is stable and can be built ? and run on my local machine ?

**Reply by primetomas** [reply]
This discussion is already completed. If you have some other issues, open a new discussion and provide full details of what you are doing so someone could try to run the same commands.

**Reply by SaeidKazemi78** [reply]
Alright ,thanks a bunch
